paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewState.kt

```diff
@@ -14,6 +14,7 @@ internal sealed class CustomerSheetViewState(
     open val isProcessing: Boolean,
     open val isEditing: Boolean,
     open val screen: PaymentSheetScreen,
+    open val currentSelection: PaymentSelection?,
```

Comment:
I don’t think we should be carrying this around. Instead, we should do the following:

1. Make the `BackHandler` invoke the ViewModel instead of calling `finishWithResult`. This allows us to go through `onDismissed`.
2. In `onDismissed`, we should add code to find `SelectPaymentMethod` in the back stack and use its `paymentSelection`.

Bullet 1 might be complicated due to the activity test, but seems like a worthy effort to me.